File: src/console/command/parameter/def/ServerGroupParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\CommandParameter;
use pocketcloud\cloud\console\command\parameter\exception\ArgumentParseException;
use pocketcloud\cloud\group\ServerGroup;
use pocketcloud\cloud\group\ServerGroupManager;

readonly class ServerGroupParameter extends CommandParameter {

    public function parseValue(string $input): ServerGroup {
        if (($group = ServerGroupManager::getInstance()->get($input)) !== null) return $group;
        throw new ArgumentParseException();
    }

    public function onTabCompleteMatch(string $currentArg): array {
        return array_filter(array_keys(ServerGroupManager::getInstance()->getAll()), fn(string $group) => str_contains(strtolower($group), $currentArg));
    }

    public function getType(): string {
        return "server";
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/parameter/def/ServerGroupParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.api.component.group.IServerGroup;
import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.exception.ArgumentParseException;
import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;

import java.util.List;

public final class ServerGroupParameter extends BaseCommandParameter {

    public ServerGroupParameter(String name, boolean optional) {
        super(name, optional);
    }

    public ServerGroupParameter(String name, boolean optional, String customErrorMessage) {
        super(name, optional, customErrorMessage);
    }

    @Override
    public Object parseValue(String input) throws ArgumentParseException {
        return PocketCloud.instance().serverGroups().get(input).orElseThrow(() -> new ArgumentParseException("ServerGroup not found"));
    }

    @Override
    public List<String> onTabCompleteMatch(String currentArg) {
        return PocketCloud.instance().serverGroups().getAll().stream().map(IServerGroup::name).toList();
    }

    @Override
    public String getType() {
        return "server_group";
    }
}